src/main/scala/com/hortonworks/spark/sql/hive/llap/FilterPushdown.scala

```diff
@@ -26,6 +26,25 @@ private[llap] object FilterPushdown extends Object {
     if (filterExpressions.isEmpty) "" else "WHERE " + filterExpressions
   }
 
+  def supportedFilter(filter: Filter) : Boolean = filter match {
+    case  EqualTo(_, _) |
+          EqualNullSafe(_, _) |
+          LessThan (_, _) |
+          GreaterThan (_, _) |
+          LessThanOrEqual (_, _) |
+          GreaterThanOrEqual (_, _) |
+          In (_, _) |
+          StringStartsWith (_, _) |
+          StringEndsWith (_, _) |
+          StringContains (_, _) |
+          IsNull (_) |
+          IsNotNull (_) |
+          And (_, _) |
+          Or (_, _) |
+          Not (_) => true
```

Comment:
@EricWohlstadter, seems it's not going to support nested filters.

Reply:
Can we maybe, instead, use `buildFilterExpression(...).isDefined`?

Reply:
@HyukjinKwon 

Good idea. I made that change.